drivers/mge-hid.c

```diff
@@ -191,9 +192,28 @@ static char		mge_scratch_buf[20];
  * discharging.
  */
 
-#define			ABM_UNKNOWN  -1
-#define			ABM_DISABLED  0
-#define			ABM_ENABLED   1
+#define			ABM_UNKNOWN     -1
+#define			ABM_DISABLED     0
+#define			ABM_ENABLED      1
+
+/* Some untis like Eaton 9E Models that using "x.ChargerType=5" and "x.Status=2"
+ * instead of other units that has ABM by "x.ABMEnable and "x.Mode".
+ * NOTE: Eaton 5P unit has "x.ABMEnable=1 and "x.Mode=4","x.ChargerType=4" and "x.Status=19?" */
+#define			ABM_ENABLED_TYPE 4
+
+/* Define if we have battery.charger.type.status for 9E Models and others */
+#define			ABM_CHARGER_NO_STATUS -1
+#define			ABM_CHARGER_STATUS 1
+
+/* Define if we have battery.charger.mode.staus by "x.ABMEnable and "x.Mode" */
+#define			ABM_CHARGER_NO_MODE -1
+#define			ABM_CHARGER_MODE 1
+
+/* Internal flag to process battery.charger.type.status for 9E Models and others */
+static int advanced_battery_status = ABM_CHARGER_NO_STATUS;
+
+/* Internal flag to process battery.charger.mode.status by "x.ABMEnable and "x.Mode" */
+static int advanced_battery_mode = ABM_CHARGER_NO_MODE;
 
 /* Internal flag to process battery status (CHRG/DISCHRG) and ABM */
 static int advanced_battery_monitoring = ABM_UNKNOWN;
```

Comment:
I know this is not your code, but maybe also better to start here with `-1` and drop the `ABM_UNKNOWN` (= -1) definition for similar reason as in other comment (separation from HID/device-returned values).